### PR TITLE
Pixels enum

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Use the enum value

# Testing this PR

Check that the build is green and the tests remain green

# Related reading

https://trello.com/c/w9GnPkQ7/72-tidy-up-enum-string-constants

cc @mtbc 

first round of cleaning

### CODE REVIEW

**Comment by joshmoore** [comment]
I did a first quick read-through looking for copy-n-paste errors without seeing any.

File: components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java [hunk 1]
```diff
@@ -273,9 +274,9 @@ protected Pixels pixels() {
         pixels.setSizeZ(1);
         pixels.setSizeT(1);
         pixels.setSizeC(1);
-        pixels.setSha1("UNKNOWN");
-        pixels.setPixelsType(new PixelsType("int8"));
-        pixels.setDimensionOrder(new DimensionOrder("XYZCT"));
+        pixels.setSha1(ChecksumAlgorithm.VALUE_ADLER_32);
```
Comment (joshmoore) [inline components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: Is this what you intended? `UNKNOWN` --> `Adler-32`?

Comment (jburel) [reply components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: it is, I assume that it was similar to "pending..." we have in various locations
Happy to revert if we want to test the insert of such value

Comment (joshmoore) [reply components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: It is like "pending..." (and therefore needs to be cleaned up / rationalized) but I think changing it to another, unexpected value ("Adler-32") is probably out of scope of this PR.

Comment (jburel) [reply components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: The adler-32 is due to code completion in eclipse :-) I will pick another value

File: components/tools/OmeroJava/test/integration/ModelMockFactory.java [hunk 2]
```diff
@@ -86,16 +86,16 @@
     static int SIZE_T = 10;
 
     /** The dimension order for the pixels type. */
-    static String XYZCT = "XYZCT";
+    static String XYZCT = ome.model.enums.DimensionOrder.VALUE_XYZCT;
```
Comment (mtbc) [inline components/tools/OmeroJava/test/integration/ModelMockFactory.java]: These classes should already have access to constants like `omero.model.enums.DimensionOrderXYZCT.value`.

**Comment by jburel** [comment]
@mtbc should be sorted now

File: components/blitz/src/loci/ome/io/OmeroReader.java [hunk 3]
```diff
@@ -369,7 +369,7 @@ else if (key.equals("iid")) {
             m.sizeT = sizeT;
             m.rgb = false;
             m.littleEndian = false;
-            m.dimensionOrder = "XYZCT";
+            m.dimensionOrder = ome.model.enums.DimensionOrder.VALUE_XYZCT;
```
Comment (mtbc) [inline components/blitz/src/loci/ome/io/OmeroReader.java]: should be able to use `omero.model.*` here

File: components/blitz/src/ome/formats/OMEROMetadataStoreClient.java [hunk 4]
```diff
@@ -5695,7 +5695,7 @@ public void setPixelsDimensionOrder(
         // We're always the same dimension order in the server; force it to
         // "XYZCT" (ticket:3124, ticket:3718, ticket:3668)
         o.setDimensionOrder((DimensionOrder) getEnumeration(
-            DimensionOrder.class, "XYZCT"));
+            DimensionOrder.class, ome.model.enums.DimensionOrder.VALUE_XYZCT));
```
Comment (mtbc) [inline components/blitz/src/ome/formats/OMEROMetadataStoreClient.java]: should be able to use `omero.model.*` here

File: components/blitz/src/ome/formats/importer/ImportSize.java [hunk 5]
```diff
@@ -81,17 +81,17 @@ public ImportSize(String fileName, Pixels pixels, String dimOrder) {
     }
 
     private int getSequenceNumber(String dimOrder) {
-        if (dimOrder.equals("XYZTC"))
+        if (ome.model.enums.DimensionOrder.VALUE_XYZTC.equals(dimOrder))
```
Comment (mtbc) [inline components/blitz/src/ome/formats/importer/ImportSize.java]: should be able to use `omero.model.*` here

**Comment by mtbc** [comment]
Looking through, I think there are still a fair few places where the more user-familiar `omero.model.*` are available, even beyond where I commented.

File: components/rendering/src/omeis/providers/re/Renderer.java [hunk 6]
```diff
@@ -82,18 +83,17 @@
...
+    public static final String MODEL_RGB = RenderingModel.VALUE_RGB;
 
     /** Identifies the type used to store model values. */
     public static final String MODEL_HSB = MODEL_RGB;
 
     /** Identifies the type used to store photometric interpretation values. */
-    public static final String PHOTOMETRIC_MONOCHROME = "Monochrome";
-
-    //static final String RGB_COLOR_DOMAIN = "RGB";
+    @Deprecated
```
Comment (mtbc) [inline components/rendering/src/omeis/providers/re/Renderer.java:95]: :+1: for deprecations

File: components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java [hunk 7]
```diff
@@ -273,9 +274,9 @@ protected Pixels pixels() {
         pixels.setSizeZ(1);
         pixels.setSizeT(1);
         pixels.setSizeC(1);
-        pixels.setSha1("UNKNOWN");
-        pixels.setPixelsType(new PixelsType("int8"));
-        pixels.setDimensionOrder(new DimensionOrder("XYZCT"));
+        pixels.setSha1(ChecksumAlgorithm.VALUE_SHA1_160);
```
Comment (mtbc) [inline components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: might want to leave as unknown unless the hash matches SHA1?

**Comment by jburel** [comment]
updated

**Comment by mtbc** [comment]
Looks good though still curious about https://github.com/openmicroscopy/openmicroscopy/pull/5260#discussion_r113399024.

File: components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java (hunk 7 above)
Comment (jburel) [reply components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: it is only used here as an entry in the DB

Comment (joshmoore) [reply components/server/test/ome/server/itests/pixeldata/PersistentEventLogLoaderTest.java]: Ah, ok. But if so, why not keep it as "UNKNOWN" or "TestValue" if you prefer?

**Review by mtbc — APPROVED** [review]

**Pull request merged**